jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/IPC.java

```diff
@@ -96,22 +101,27 @@ public class IPC {
 	 */
 	public static final boolean useFileLockWatchdog;
 
+	static final Properties internalProperties = VM.getVMLangAccess().internalGetProperties();
```

Comment:
System properties can be replaced via System.setProperties() so this is caching the original properties.

Reply:
`VM.getVMLangAccess().internalGetProperties()` returns `System.systemProperties`, not a copy.

Reply:
I'm not saying it's a copy, I'm saying System.systemProperties can change (be replaced) and then this code won't be using the latest system properties.

Reply:
Right, unless System.setProperties() is called, and then the system properties cached in IPC will be obsolete and not reflecting the latest system properties.

Reply:
Restored most of `VM.getVMLangAccess().internalGetProperties()` usages except those within  `AttachHandler.initialize()` which shares the returning `internalProperties` object inside the method body.

Reply:
You could add a VM or IPC internalGetProperties() helper if you wanted.

Reply:
Added `VM.internalGetProperties()` helper.